src/sentry/integrations/utils/code_mapping.py

```diff
@@ -15,6 +16,16 @@
 # We only care about the ones that would show up in stacktraces after symbolication
 EXTENSIONS = ["js", "jsx", "tsx", "ts", "mjs", "py", "rb", "php", "go"]
 
+# List of file paths prefixes that should become stack trace roots
+FILE_PATH_PREFIX_REGEX = [
+    "app:///",
+    r"[a-zA-Z]:[\\\\]+Users[\\\\]+[a-zA-Z0-9_]+[\\\\]+",
+    r"[a-zA-Z]:[\\\\]+",
+    r"\./",
+    r"[\.\./]+",
+    "~/",
+]
```

Comment:
Made a list containing the possible path prefixes

Reply:
Just curious, how do we know this is a comprehensive set?

Reply:
No, it's the ones I thought made the most sense from this [query](https://sentry.io/organizations/sentry/discover/homepage/?field=stacktrace_link.file&field=count%28%29&field=http.referer&field=stacktrace_link.in_app&name=All+Events&project=1&query=transaction%3A%2Fapi%2F0%2Fprojects%2F%7Borganization_slug%7D%2F%7Bproject_slug%7D%2Fstacktrace-link%2F+stacktrace_link.platform%3Anode&sort=-stacktrace_link.file&statsPeriod=14d&yAxis=count%28%29), but I think I'll need some feedback from Armen to make sure it's comprehensive/correct

Reply:
Unfortunately, using `stacktrace_link.file` as a source for this is a bit misleading since you will have values of what customers are doing during development and before applying source maps (minification). You can use ` stacktrace_link.found:True` to find better samples.

To understand minification, look at this:
* `webpack://backend/src/shared/foo.ts` -> Raw path; no minification.
* `/usr/build/bar.js` -> After minification (the path does not have to resemble the original raw path)

Once you load a sample event, use this in the UI to change between the values.
<img width="510" alt="image" src="https://user-images.githubusercontent.com/44410/210808714-79008a0f-6fde-43ec-8428-956926c963be.png">

If you see any paths starting with `webpack://` you will know that we have not had minification happen (lack of source maps).

## Recommendation

For now, let's remove the Windows logic and see what other changes we will need to remove. I do not think we need the regex logic but it will be easier to see once we remove the Windows support from this PR.

## New matching logic

While I was doing some analysis (see below), I believe we need a modification on how we search for files in the trees. We should put this PR on hold and implement the following.

Given, `/var/foo/bar/baz.js`, search for `baz.js` in the trees, if more than one match is found try the same path but include an extra dir (if available, otherwise, stop). With an extra path search for `bar/baz.js`, try again until you only find one match or fail to do so.

In the case, I'm looking the stack trace root would be `/var/` and the source code root is "".

In Python, we kind of already do this. We just assume that there's going to be one level of stack trace root and try to look for the full path without `/var/`. See these:
https://github.com/getsentry/sentry/blob/37959551c740baf289adcc39c7f95fc42802004a/src/sentry/integrations/utils/code_mapping.py#L338-L340
https://github.com/getsentry/sentry/blob/37959551c740baf289adcc39c7f95fc42802004a/src/sentry/integrations/utils/code_mapping.py#L344

This means that we would not need two different matching logics (one for packaged languages and one for straight path logic). It would also increase the likelyhood of finding code mappings since the logic to calculate `file_and_dir_path` is error prone.

## Analysis notes
The question is, what will we see in the ingestion pipeline?

If we load a JSON event for a Node error where there is minification happening, I can see the following:
<img width="396" alt="image" src="https://user-images.githubusercontent.com/44410/210814379-5750080b-d7bb-4b27-b926-132120b51ba4.png">

In this case, we would want to use the unminified or unsymbolicated path: `"../../../src/lib/daily-dive/data-element.js"`. We would want to drop the `../` from the path and then search in the trees for it.

In some case, you will see paths starting with `app:///` and you would want to drop it from the path before searching for the file in the trees.

In other cases, you will get something like `/var/task/foo/bar/baz.js`, you will want to first search for `baz.js` and if there's more t